boards/msp_exp432p401r/src/main.rs

```diff
@@ -0,0 +1,163 @@
+#![no_std]
+#![cfg_attr(not(doc), no_main)]
+#![feature(asm, core_intrinsics)]
+// #![deny(missing_docs)]
+
+use kernel::capabilities;
+use kernel::common::dynamic_deferred_call::DynamicDeferredCall;
+use kernel::common::dynamic_deferred_call::DynamicDeferredCallClientState;
+use kernel::component::Component;
+use kernel::hil::watchdog::Watchdog;
+use kernel::Platform;
+use kernel::{create_capability, debug, static_init};
+
+pub mod io;
+
+// Number of concurrent processes this platform supports.
+const NUM_PROCS: usize = 4;
+
+// Actual memory for holding the active process structures.
+static mut PROCESSES: [Option<&'static dyn kernel::procs::ProcessType>; NUM_PROCS] =
+    [None; NUM_PROCS];
+
+// Static reference to chip for panic dumps.
+static mut CHIP: Option<&'static msp432::chip::Msp432> = None;
+
+// How should the kernel respond when a process faults.
+const FAULT_RESPONSE: kernel::procs::FaultResponse = kernel::procs::FaultResponse::Panic;
+
+// RAM to be shared by all application processes.
+#[link_section = ".app_memory"]
+static mut APP_MEMORY: [u8; 32768] = [0; 32768];
+
+// Force the emission of the `.apps` segment in the kernel elf image
+// NOTE: This will cause the kernel to overwrite any existing apps when flashed!
+#[used]
+#[link_section = ".app.hack"]
+static APP_HACK: u8 = 0;
```

Comment:
Do you mind explaining why you need this? Can apps not be flashed to the board?

Reply:
I implemented it like the stm32f3discovery board, where the applications are flashed combined with the kernel. For me this seemed to be the easiest method to get the platform working at all. Is there a better/easier method for doing this?

Reply:
Most other boards allow you to flash the kernel and apps separately.You can just basically run the same flash command but with different addresses.

Something like this: https://github.com/tock/tock/blob/76b00caa8a7bb54778dcbbcdc9a1ce38148ef1fc/boards/nano33ble/README.md#programming-applications

Reply:
Ah, I see!
Should I change this that I can flash the apps separately with openOCD (I guess this should be possible)? Or should I keep as it is now?

Reply:
Unless there is a reason you can't (the HW won't let you) you should allow flashing them separately. You can add a `make flash-app` target in the `Makefile` to make it easier to do if you want.

Reply:
Ok, I added a new target in the makefile. Now it looks like the following:
```make
# Makefile for building the tock kernel for the MSP-EXP432P401 launchpad
#
TARGET=thumbv7em-none-eabi
PLATFORM=msp-exp432p401r

include ../Makefile.common

OPENOCD=openocd
OPENOCD_OPTIONS=-f openocd.cfg
APP_START_ADDR=0x20000
APP?=

.PHONY: flash-debug
flash-debug: $(TOCK_ROOT_DIRECTORY)target/$(TARGET)/debug/$(PLATFORM).elf
	$(OPENOCD) $(OPENOCD_OPTIONS) -c "init; reset halt; flash write_image erase $<; verify_image $<; reset; shutdown"

.PHONY: flash
flash: $(TOCK_ROOT_DIRECTORY)target/$(TARGET)/release/$(PLATFORM).elf
	$(OPENOCD) $(OPENOCD_OPTIONS) -c "init; reset halt; flash write_image erase $<; verify_image $<; reset; shutdown"

.PHONY: flash-app
flash-app:
	$(OPENOCD) $(OPENOCD_OPTIONS) -c "init; reset halt; flash write_image erase $(APP) $(APP_START_ADDR) bin; verify_image_checksum $(APP) $(APP_START_ADDR) bin; reset; shutdown"

.PHONY: program
program: $(TOCK_ROOT_DIRECTORY)target/$(TARGET)/debug/$(PLATFORM).elf
	$(error See README.md and update this section accordingly)`

Can I keep it like that? Or should I do something in a different way?